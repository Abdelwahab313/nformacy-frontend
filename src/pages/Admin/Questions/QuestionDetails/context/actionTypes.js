const QuestionActionTypes = {
  UPDATE_QUESTION: 'UPDATE_QUESTION',
};

export default QuestionActionTypes;
