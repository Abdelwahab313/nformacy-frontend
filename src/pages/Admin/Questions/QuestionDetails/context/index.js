import React, { createContext, useContext, useReducer } from 'react';
import QuestionActionTypes from './actionTypes';

const defaultValues = {
  questionDetails: '',
};

const QuestionContext = createContext();

const QuestionProvider = ({ children, initialValue = {} }) => {
  initialValue = { ...defaultValues, ...initialValue };
  const { Provider } = QuestionContext;
  return (
    <Provider value={useReducer(QuestionReducer, initialValue)}>
      {children}
    </Provider>
  );
};

const useQuestionContext = () => {
  const context = useContext(QuestionContext);
  if (context === undefined) {
    throw new Error('useQuestionContext must be used within a AuthProvider');
  }
  return context;
};

const QuestionReducer = (state, action) => {
  switch (action.type) {
    case QuestionActionTypes.UPDATE_QUESTION:
      const updatedQuestion = { ...state.questionDetails, ...action.payload };
      return { ...state, questionDetails: updatedQuestion };

    default:
      throw new Error(`Unhandled action type: ${action.type}`);
  }
};

export { useQuestionContext, QuestionProvider };
