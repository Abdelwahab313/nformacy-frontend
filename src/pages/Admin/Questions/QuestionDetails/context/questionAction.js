import QuestionActionTypes from './actionTypes';

export const updateQuestionDetails = (dispatch, question) => {
  dispatch({
    type: QuestionActionTypes.UPDATE_QUESTION,
    payload: question,
  });
};

export const setErrorMessage = (dispatch, message) => {
  dispatch({
    type: QuestionActionTypes.UPDATE_ERROR_MESSAGE,
    payload: message,
  });
};

export const setSuccessMessage = (dispatch, message) => {
  dispatch({
    type: QuestionActionTypes.UPDATE_SUCCESS_MESSAGE,
    payload: message,
  });
};

export const setEmptyMessage = (dispatch) => {
    dispatch({
      type: QuestionActionTypes.EMPTY_MESSAGE,
    });
  };