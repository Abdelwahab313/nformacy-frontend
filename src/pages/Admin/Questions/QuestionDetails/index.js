import React, { useCallback, useEffect, useState } from 'react';
// @material-ui/core components
// core components
import GridItem from 'components/Grid/GridItem.js';
import GridContainer from 'components/Grid/GridContainer.js';
import Button from 'components/buttons/RegularButton.js';
import Card from 'components/Card/Card.js';
import CardHeader from 'components/Card/CardHeader.js';
import CardFooter from 'components/Card/CardFooter.js';
import { useHistory, useLocation } from 'react-router';
import { fetchQuestionDetails } from 'apis/questionsAPI';
import LoadingCircle from 'components/progress/LoadingCircle';
import { approveQuestion } from '../../../../apis/questionsAPI';
import QuestionForm from './subComponents/QuestionForm';
import { useStyles } from '../../../../styles/Admin/questionFormStyles';
import AnswerView from './subComponents/AnswerView';
import authManager from '../../../../services/authManager';
import { QuestionProvider, useQuestionContext } from './context';
import { Typography } from '@material-ui/core';
import { updateQuestionDetails } from './context/questionAction';

const QuestionDetailsPage = () => {
  const classes = useStyles();
  const [{ questionDetails }, dispatch] = useQuestionContext();
  const [isLoadingForUpdating, setIsLoadingForUpdating] = useState(false);
  const [isLoading, setIsLoading] = useState(false);
  let history = useHistory();

  const location = useLocation();
  const questionId = location?.state?.questionId;
  const isNewQuestion = !questionId;

  useEffect(() => {
    setIsLoading(true);
    fetchQuestionDetails(questionId)
      .then((response) => {
        updateQuestionDetails(dispatch, response.data);
      })
      .finally(() => {
        setIsLoading(false);
      });
  }, []);

  const setRating = useCallback(
    (answerIndex, rating) => {
      const newQuestionDetails = { ...questionDetails };
      newQuestionDetails.answers[answerIndex].rating = rating;
      updateQuestionDetails(dispatch, newQuestionDetails);
    },
    [questionDetails, dispatch],
  );

  if (isLoading) {
    return <LoadingCircle />;
  }

  const onDeployQuestionClicked = () => {
    setIsLoadingForUpdating(true);
    approveQuestion(questionDetails.id)
      .then(() => {
        history.push('/admin/questions');
      })
      .catch(() => {})
      .finally(() => setIsLoadingForUpdating(false));
  };

  return (
    <GridContainer justifyContent={'center'}>
      <GridItem xs={12} sm={12} md={12}>
        <Card>
          <CardHeader color='primary'>
            <Typography component={'h4'} id={'post-question-page-header'}>
              {isNewQuestion ? 'Add Question' : 'Edit Question'}
            </Typography>
          </CardHeader>
          <QuestionForm isNewQuestion={isNewQuestion} />
          <CardFooter className={classes.footerButtons}>
            {!isNewQuestion &&
              authManager.isAdmin() &&
              questionDetails.state === 'pending_deployment_to_roaster' && (
                <Button
                  id={'approveQuestion'}
                  disabled={isLoadingForUpdating}
                  onClick={onDeployQuestionClicked}
                  color='primary'>
                  Deploy to question roaster
                </Button>
              )}
          </CardFooter>
        </Card>
      </GridItem>
      {!isNewQuestion && questionDetails.answers && (
        <GridItem xs={12}>
          {questionDetails.answers?.map((answer, index) => (
            <div id={answer.referenceNumber} key={`answer-${index}`}>
              <AnswerView answer={answer} index={index} setRating={setRating} />
            </div>
          ))}
        </GridItem>
      )}
    </GridContainer>
  );
};

const QuestionDetails = (props) => {
  return (
    <QuestionProvider>
      <QuestionDetailsPage {...props} />
    </QuestionProvider>
  );
};
export default QuestionDetails;
