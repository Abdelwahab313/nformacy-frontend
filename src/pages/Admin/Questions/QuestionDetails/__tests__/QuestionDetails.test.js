import { render, waitFor } from '@testing-library/react';
import { AuthProvider } from '../../../../auth/context/auth';
import { LocaleProvider } from '../../../../../hooks/localization/context';
import { fetchQuestionDetails } from '../../../../../apis/questionsAPI';
import React from 'react';
import QuestionDetails from '../index';
import { Router } from 'react-router';
import { createMemoryHistory } from 'history';

jest.mock('apis/questionsAPI', () => ({
  __esModule: true, // this property makes it work
  default: 'mockedDefaultExport',
  fetchQuestionDetails: jest.fn().mockResolvedValue({
    data: {
      assignedAdviserId: 2,
      assignmentType: 'project',
      content:
        'From the viewpoint of what you can do, therefore, languages do differ - but the differences are limited. For example, Python and Ruby provide almost the same power to the programmer.',
      createdAt: '2020-09-14T09:25:55.689Z',
      currentActionTime: '2020-09-14T22:02:28.411Z',
      field: [{ value: 'marketingAndPR', label: 'Marketing and PR' }],
      hoursToCloseAnswers: null,
      hoursToReviewAndEdit: null,
      id: 1,
      industry: { value: 'consulting', label: 'Consulting' },
      isApproved: true,
      mediaId: null,
      referenceNumber: 2000100,
      state: 'pending_adviser_acceptance',
      subfield: [{ value: 'marketResearch', label: 'Market Research' }],
      title: 'Elmer',
      answers: [
        {
          content: 'test content',
        },
      ],
    },
  }),
}));


jest.mock('apis/fieldsAPI', () => ({
  __esModule: true, // this property makes it work
  default: 'mockedDefaultExport',
  fetchFields: jest.fn().mockResolvedValue({
    data: [
      {
        id: 1,
        label: 'Finance',
      },
      {
        id: 2,
        label: 'Marketing',
      },
      {
        id: 3,
        label: 'FinTech',
      },
      {
        id: 4,
        label: 'Accounting',
      },
    ],
  }),
}));

it('should match snapshot', async () => {
  const history = createMemoryHistory();
  history.push('/', { questionId: 1 });
  const { asFragment } = render(
    <Router history={history}>
      <AuthProvider>
        <LocaleProvider initialLocale={'en'}>
          <QuestionDetails />
        </LocaleProvider>
      </AuthProvider>
    </Router>,
  );
  await waitFor(() => expect(fetchQuestionDetails).toHaveBeenCalledTimes(1));
  expect(asFragment()).toMatchSnapshot();
});
