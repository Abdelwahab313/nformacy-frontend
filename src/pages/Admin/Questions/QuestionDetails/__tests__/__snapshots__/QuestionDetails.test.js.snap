// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root grid MuiGrid-container"
    justifycontent="center"
  >
    <div
      class="MuiGrid-root grid MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12"
    >
      <div
        class="card"
      >
        <div
          class="cardHeader primaryCardHeader"
        >
          <h4
            class="MuiTypography-root MuiTypography-body1"
            id="post-question-page-header"
          >
            Edit Question
          </h4>
        </div>
        <div
          class="cardBody"
        >
          <div
            class="MuiGrid-root grid MuiGrid-container"
          >
            <div
              class="MuiGrid-root grid MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-2"
            >
              <div
                class="MuiFormControl-root undefined formControl MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root labelRootundefined MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                  data-shrink="true"
                  for="reference-id"
                >
                  Reference ID
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline underline Mui-disabled Mui-disabled disabled MuiInputBase-formControl MuiInput-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled disabled"
                    disabled=""
                    id="reference-id"
                    name="referenceNumber"
                    type="text"
                    value="2000100"
                  />
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root grid MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-7"
            >
              <div
                class="MuiFormControl-root undefined formControl MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root labelRootundefined MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                  data-shrink="true"
                  for="title"
                >
                  Title
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline underline MuiInputBase-formControl MuiInput-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input"
                    id="title"
                    name="title"
                    type="text"
                    value="Elmer"
                  />
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root grid MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-3"
            >
              <div
                class="MuiFormControl-root undefined formControl MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root labelRootundefined MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                  data-shrink="true"
                  for="post-date"
                >
                  Post Date
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline underline Mui-disabled Mui-disabled disabled MuiInputBase-formControl MuiInput-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled disabled"
                    disabled=""
                    id="post-date"
                    name="createdAt"
                    type="text"
                    value="1 month ago"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root grid inputsRow MuiGrid-container"
          >
            <div
              class="MuiGrid-root grid MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-3"
            >
              <div
                aria-expanded="false"
                class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
                name="majorFieldsOfExperience"
                role="combobox"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                    data-shrink="false"
                    for="majorFieldsOfExperienceSelect"
                    id="majorFieldsOfExperienceSelect-label"
                  >
                    Major Fields Of Experience
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot makeStyles-inputRoot-6 MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                  >
                    <input
                      aria-autocomplete="list"
                      aria-invalid="false"
                      autocapitalize="none"
                      autocomplete="off"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                      id="majorFieldsOfExperienceSelect"
                      spellcheck="false"
                      type="text"
                      value=""
                    />
                    <div
                      class="MuiAutocomplete-endAdornment"
                    >
                      <button
                        aria-label="Clear"
                        class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
                        tabindex="-1"
                        title="Clear"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                      <button
                        aria-label="Open"
                        class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                        tabindex="-1"
                        title="Open"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M7 10l5 5 5-5z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-9"
                      >
                        <span>
                          Major Fields Of Experience
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root grid MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-3"
            >
              <div
                aria-expanded="false"
                class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
                name="fields"
                role="combobox"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                    data-shrink="false"
                    for="specificFieldsOfExperienceSelect"
                    id="specificFieldsOfExperienceSelect-label"
                  >
                    Specific Fields Of Experience
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot makeStyles-inputRoot-6 MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                  >
                    <input
                      aria-autocomplete="list"
                      aria-invalid="false"
                      autocapitalize="none"
                      autocomplete="off"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                      id="specificFieldsOfExperienceSelect"
                      spellcheck="false"
                      type="text"
                      value=""
                    />
                    <div
                      class="MuiAutocomplete-endAdornment"
                    >
                      <button
                        aria-label="Clear"
                        class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
                        tabindex="-1"
                        title="Clear"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                      <button
                        aria-label="Open"
                        class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                        tabindex="-1"
                        title="Open"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M7 10l5 5 5-5z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-9"
                      >
                        <span>
                          Specific Fields Of Experience
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root grid MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-3"
            >
              <div
                aria-expanded="false"
                class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
                name="industry"
                role="combobox"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                    data-shrink="true"
                    for="industry"
                    id="industry-label"
                  >
                    Industry
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                  >
                    <input
                      aria-autocomplete="list"
                      aria-invalid="false"
                      autocapitalize="none"
                      autocomplete="off"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                      id="industry"
                      spellcheck="false"
                      type="text"
                      value="Consulting"
                    />
                    <div
                      class="MuiAutocomplete-endAdornment"
                    >
                      <button
                        aria-label="Clear"
                        class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator MuiAutocomplete-clearIndicatorDirty"
                        tabindex="-1"
                        title="Clear"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                      <button
                        aria-label="Open"
                        class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                        tabindex="-1"
                        title="Open"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M7 10l5 5 5-5z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-9 PrivateNotchedOutline-legendNotched-10"
                      >
                        <span>
                          Industry
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root grid MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-3"
            >
              <div
                aria-expanded="false"
                class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
                name="QuestionLanguage"
                role="combobox"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                    data-shrink="false"
                    for="questionLanguage"
                    id="questionLanguage-label"
                  >
                    Question Language
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                  >
                    <input
                      aria-autocomplete="list"
                      aria-invalid="false"
                      autocapitalize="none"
                      autocomplete="off"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                      id="questionLanguage"
                      spellcheck="false"
                      type="text"
                      value=""
                    />
                    <div
                      class="MuiAutocomplete-endAdornment"
                    >
                      <button
                        aria-label="Clear"
                        class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator MuiAutocomplete-clearIndicatorDirty"
                        tabindex="-1"
                        title="Clear"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                      <button
                        aria-label="Open"
                        class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                        tabindex="-1"
                        title="Open"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M7 10l5 5 5-5z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-9"
                      >
                        <span>
                          Question Language
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root grid inputsRow MuiGrid-container"
          >
            <div
              class="MuiGrid-root grid MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root contentTitle MuiInputLabel-animated"
              >
                Question Content
              </label>
              <div
                class="MuiGrid-root questionContainer MuiGrid-container"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <textarea
                    id="richContent"
                    style="visibility: hidden;"
                  />
                </div>
                <div
                  class="MuiGrid-root undefined attachmentContainer MuiGrid-item MuiGrid-grid-xs-6"
                >
                  <div
                    class="attachmentUploaderContainer"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-flex-start"
                    >
                      <div
                        class="fileContainer"
                        style="box-shadow: none; padding: 0px; margin: 0px 10px 0px 0px;"
                      >
                        <div
                          class="errorsContainer"
                        />
                        <button
                          class="MuiButtonBase-root MuiButton-root MuiButton-contained"
                          id="chooseFileButton"
                          style="background-color: rgb(18, 87, 115); color: rgb(255, 255, 255);"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="MuiButton-label"
                          >
                            Attach
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                        <input
                          accept=".doc,.docx,.xml,image/*,video/*,.pdf,.csv"
                          name=""
                          type="file"
                        />
                      </div>
                      <div
                        class="flipMove"
                        style="position: relative;"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root answerButtonContainer MuiGrid-item MuiGrid-grid-xs-6"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained answerButtons"
                    id="applyChangesButton"
                    style="background-color: rgb(18, 87, 115); color: rgb(255, 255, 255);"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Apply Changes
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="cardFooter footerButtons"
        />
      </div>
    </div>
    <div
      class="MuiGrid-root grid MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div>
        <div
          class=""
        >
          <div
            class="MuiGrid-root grid MuiGrid-container"
          >
            <div
              class="MuiGrid-root grid answerRowStyles MuiGrid-item MuiGrid-grid-xs-12"
            >
              <span
                aria-label="null Stars"
                class="MuiRating-root MuiRating-readOnly"
                role="img"
              >
                <span>
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </span>
                </span>
                <span>
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </span>
                </span>
                <span>
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </span>
                </span>
                <span>
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </span>
                </span>
                <span>
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </span>
                </span>
              </span>
            </div>
            <div
              class="MuiGrid-root grid answerRowStyles MuiGrid-item MuiGrid-grid-xs-2"
            />
            <div
              class="MuiGrid-root grid answerRowStyles MuiGrid-item MuiGrid-grid-xs-10"
            >
              <div
                class="MuiGrid-root MuiGrid-container"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Invalid Date
                </p>
              </div>
            </div>
            <div
              class="MuiGrid-root grid answerRowStyles MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiGrid-root MuiGrid-container"
              >
                <p
                  class="MuiTypography-root answerFieldLabel MuiTypography-body1"
                >
                  Consultant:
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1"
                />
              </div>
            </div>
            <div
              class="MuiGrid-root grid answerRowStyles MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiGrid-root answerContent MuiGrid-item"
              >
                <div>
                  <span
                    width="0"
                  >
                    <span />
                    <span>
                      <div>
                        test content
                      </div>
                    </span>
                    <span
                      style="position: fixed; visibility: hidden; top: 0px; left: 0px;"
                    >
                      <span>
                        ... 
                        <a
                          class="anchor"
                          href="#"
                        >
                          <div
                            class="MuiGrid-root icons MuiGrid-container"
                          >
                            <div>
                               showMore 
                            </div>
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root icon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                              />
                            </svg>
                          </div>
                        </a>
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root grid answerRowStyles MuiGrid-item MuiGrid-grid-xs-12"
            />
            <div
              class="MuiGrid-root grid answerRowStyles MuiGrid-item MuiGrid-grid-xs-2"
            />
            <div
              class="MuiGrid-root grid MuiGrid-item MuiGrid-grid-xs-10"
            />
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row-reverse MuiGrid-align-items-xs-flex-end"
            />
          </div>
          <hr
            class="MuiDivider-root divider MuiDivider-middle"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
